tools/cli/main.js

```diff
@@ -768,33 +768,40 @@ Fiber(function () {
     rawArgs.push(term);
   }
 
-  // Prevent running meteor as root
-  if (process.getuid) {
-    // On UNIX platforms.
-    if (process.getuid() === 0){
-      if (_.has(rawOptions, '--unsafe-perm')) {
-        // Meteor is running as root and has the --unsafe-perm flag, just notice.
-        Console.info("");
-        Console.info("You have run Meteor as root.",
-        "Your permissions in your app directory will be incorrect if you ever attempt to perform any Meteor tasks as your non-root user.",
-        "You probably didn't want this, but you can fix it by running the following from the root of your project:");
-        Console.info("");
-        Console.info("sudo chown -Rh <username> .meteor/local");
-        Console.info("");
-        // Fix issue when running `sudo meteor --unsafe-perm`, --unsafe-perm: unknown option.
-        delete rawOptions['--unsafe-perm'];
-      }
-      else {
-        // Meteor is running as root without the --unsafe-perm flag, notice and stop.
-        Console.error("");
-        Console.error("You are attempting to run Meteor as the \"root\" user.",
+  if (_.has(rawOptions, "--unsafe-perm")) {
+    process.env.METEOR_UNSAFE_PERM = "true";
+    delete rawOptions["--unsafe-perm"];
+  }
+
+  // Prevent running meteor as root on UNIX platforms.
+  if (process.getuid &&
+      process.getuid() === 0) {
+    const unsafePerm = !! (
+      process.env.METEOR_UNSAFE_PERM &&
+      JSON.parse(process.env.METEOR_UNSAFE_PERM));
+
+    if (! unsafePerm) {
+      // Meteor is running as root without $METEOR_UNSAFE_PERM, notice and stop.
+      Console.error("");
+      Console.error(
+        "You are attempting to run Meteor as the root user.",
         "If you are developing, this is almost certainly *not* what you want to do and will likely result in incorrect file permissions.",
-        "However, if you are running this in a build process (CI, etc.) or you are absolutely sure you know what you are doing,",
-        "add the `--unsafe-perm` flag to this command to proceed.");
-        Console.error("");
+        "However, if you are running this command in a build process (CI, etc.), or you are absolutely sure you know what you are doing,",
+        "set the $METEOR_UNSAFE_PERM environment variable to proceed."
```

Comment:
Kind of relevant, and not mentioned in my quick, slumberous reply above last night was clarification as to _why_ I prefer not to use environment variables as a primary means: they are confusing and difficult to convey across different platforms (dollar signs included or not).

For example, I actually don't know how this variable affects Windows users, but the customer-support script to recommend how to set environment variables goes something like this:

> _Hi, so I need you need to set this environment variable: If you're on Windows and using PowerShell, you should run `$Env:METEOR_UNSAFE_PERM="true"`, however if you're using command prompt you should run `set METEOR_UNSAFE_PERM=true` but on Unix you'll want to run `METEOR_UNSAFE_PERM=true`._ 🤕 

It should be as easy as "You should add `--unsafe-perm` to the command"...which is why I've rallied for `--unsafe-perm` and `--no-release-check` on past PRs.  

That being said, I do think it's important to have the environment variables available in many cases (and I understand the importance/use for `meteor npm`, in this case).  Hopefully if #7559 ever comes to fruition it will support both arguments and environment variables alike, [ala `npm config`](https://docs.npmjs.com/misc/config).